Automattic/jetpack
# [RNMobile] Fix broken block registration

Fixes a regression in the Jetpack app that followed the changes in https://github.com/Automattic/jetpack/pull/32697. With this fix, it's possible to add Jetpack blocks within the apps again.  

Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6342

## Proposed changes:

* The `registerJetpackBlockFromMetadata` function has been taken from [the web's register-jetpack-block.js file](https://github.com/Automattic/jetpack/blob/f043df7738f5c7d28b4a2f6affd9e8812e314fdd/projects/plugins/jetpack/extensions/shared/register-jetpack-block.js#L80-L98) add [added to register-jetpack-block.native.js](https://github.com/Automattic/jetpack/blob/f043df7738f5c7d28b4a2f6affd9e8812e314fdd/projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js) for use in the apps.
* Some tweaks have been made to ensure the function works as expected with the apps, including the use of a custom function (`SvgXml` from [the react-native-svg library](https://github.com/software-mansion/react-native-svg)) for compatibility with React Native.

### Other information:


## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

* With these changes checked out, navigate to the editor in the iOS or Android app.
* Select + to insert a new block, verify that the following Jetpack blocks are visible: 
  * Contact Info
  * Story
  * Tiled Gallery (dev-environments only)
* Verify that it's possible to add each block and that any edits are saved as expected.

| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/Automattic/jetpack/assets/2998162/0312985e-75fc-4b0d-9e6b-d95f5f8ba24c" width="100%"> | <img src="https://github.com/Automattic/jetpack/assets/2998162/924c716b-48b4-4dec-bd3a-96d5e8987513" width="100%"> |
